File: 1461. Check If a String Contains All Binary Codes of Size K.ts
function hasAllCodes(s: string, k: number): boolean {
    
    if(s.length < k) return false;
    let set = new Set<string>();

    for(let i=0;i<=s.length-k;i++){
        set.add(s.substring(i,i+k));
    }
    return set.size === 1<<k;

};

File: 1461. Check If a String Contains All Binary Codes of Size K.cpp
class Solution {
public:
    bool hasAllCodes(string s, int k) {
       // if(s[0]=='0' && s.size()==1) return false;
        if(s.size() < k) return false;
        unordered_set<string>st;
        for(int i=0;i<=(s.size()-k);i++){
            st.insert(s.substr(i,k));
        }
        return st.size() == 1<<k;
    }
};